iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/Tools.java

```diff
@@ -124,6 +124,112 @@ public static boolean isCause(Class<? extends Throwable> possibleExceptionCause,
+        }
+    }
+
+    public static TestModuleIdentity getTestModule(String iotHubConnectionString, IotHubClientProtocol protocol, AuthenticationType authenticationType) throws URISyntaxException, IOException, IotHubException, InterruptedException, ModuleClientException, GeneralSecurityException
+    {
+        if (authenticationType == SAS)
+        {
+            //sas device to house the sas module under test
+            String deviceId = "java-method-e2e-test-device".concat("-" + UUID.randomUUID().toString());
+            Device device = Device.createFromId(deviceId, null, null);
+            device = Tools.addDeviceWithRetry(registryManager, device);
+
+            //sas module client under test
+            String moduleId = "java-method-e2e-test-module".concat("-" + UUID.randomUUID().toString());
+            Module module = Module.createFromId(deviceId, moduleId, null);
+            module = Tools.addModuleWithRetry(registryManager, module);
+            ModuleClient moduleClient = new ModuleClient(DeviceConnectionString.get(iotHubConnectionString, device, module), protocol);
+
+            return new TestModuleIdentity(moduleClient, device, module);
+        }
+        else if (authenticationType == SELF_SIGNED)
+        {
+            String x509Thumbprint = IntegrationTest.x509CertificateGenerator.getX509Thumbprint();
+            SSLContext sslContext =
+                SSLContextBuilder.buildSSLContext(
+                    IntegrationTest.x509CertificateGenerator.getPublicCertificate(),
+                    IntegrationTest.x509CertificateGenerator.getPrivateKey());
+
+            //x509 device to house the x509 module under test
+            String deviceId = "java-method-e2e-test-device-x509".concat("-" + UUID.randomUUID().toString());
+            Device device = Device.createDevice(deviceId, SELF_SIGNED);
+            device.setThumbprintFinal(x509Thumbprint, x509Thumbprint);
+            device = Tools.addDeviceWithRetry(registryManager, device);
+
+            //x509 module client under test
+            String moduleId = "java-method-e2e-test-module-x509".concat("-" + UUID.randomUUID().toString());
+            Module module = Module.createModule(deviceId, moduleId, SELF_SIGNED);
+            module.setThumbprintFinal(x509Thumbprint, x509Thumbprint);
+            module = Tools.addModuleWithRetry(registryManager, module);
+            ModuleClient moduleClient = new ModuleClient(DeviceConnectionString.get(iotHubConnectionString, device, module), protocol, sslContext);
+
+            return new TestModuleIdentity(moduleClient, device, module);
+        }
+        else
+        {
+            throw new IllegalArgumentException("Test code has not been written for this path yet");
+        }
+    }
+
+    public static void disposeTestIdentity(TestIdentity testIdentity, String iotHubConnectionString)
+    {
+        try
+        {
+            getRegistyManager(iotHubConnectionString).removeDevice(testIdentity.getDeviceId());
```

Comment:
Again, this doesn't change what we do with test identities yet, but we likely will need to start recycling these test identities rather than deleting them. For now though, no change in behavior is brought on by this PR